### PR TITLE
refactor(container-kill): Converting pumba job to pumba pod in container-kill experiment

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

**What this PR does / why we need it**:

- Converting pumba job to pumba pod in container-kill experiment

- It will retry three times for pod to come to running state before terminating the execution.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Checklist**

* [ ] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ispeakc0de** [comment]
```
2020-03-29 11:37:42.329806 ************ BRACE YOURSELF, EXPERIMENT BEGINS! ************ 

2020-03-29 11:37:45.108547 Step: Gathering Facts 
task path: /experiments/generic/container_kill/container_kill_ansible_logic.yml:2
META: ran handlers

2020-03-29 11:37:45.460019 Step: Identify the chaos util to be invoked 
task path: /experiments/generic/container_kill/container_kill_ansible_prerequisites.yml:1
details => {"changed": true, "checksum": "03b27bf248187df047c75919cd9f1b6a0d89c829", "dest": "./chaosutil.yml", "gid": 0, "group": "root", "md5sum": "9e9d63cc391e78455c2cc7892a37a36c", "mode": "0644", "owner": "root", "size": 51, "src": "/root/.ansible/tmp/ansible-tmp-1585481865.15-198024346533477/source", "state": "file", "uid": 0}

2020-03-29 11:37:45.502740 Step: include_vars 
task path: /experiments/generic/container_kill/container_kill_ansible_logic.yml:24
details => {"ansible_facts": {"c_util": "/chaoslib/pumba/pod_failure_by_sigkill.yml"}, "ansible_included_var_files": ["/experiments/generic/container_kill/chaosutil.yml"], "changed": false}

2020-03-29 11:37:45.546645 Step: Getting the chaos namespace 
task path: /experiments/generic/container_kill/container_kill_ansible_logic.yml:27
details => {"ansible_facts": {"c_ns": "shubham"}, "changed": false}

2020-03-29 11:37:45.591771 Step: Construct chaos result name 
task path: /experiments/generic/container_kill/container_kill_ansible_logic.yml:32
details => {"ansible_facts": {"c_result": "engine-container-kill"}, "changed": false}
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1

2020-03-29 11:37:45.830863 Step: [PreReq]: Generate the chaos result CR to reflect SOT (Start of Test) 
task path: /utils/runtime/update_chaos_result_resource.yml:3
details => {"changed": true, "checksum": "d0232319d3e8b88a9bc171eb4aed0887747175a4", "dest": "/utils/runtime/chaos-result.yml", "gid": 0, "group": "root", "md5sum": "25a741d10b5810b62c0a8cad83b5b86a", "mode": "0644", "owner": "root", "size": 312, "src": "/root/.ansible/tmp/ansible-tmp-1585481865.67-124439095071918/source", "state": "file", "uid": 0}

2020-03-29 11:37:46.813576 Step: [PreReq]: Apply the chaos result CR for container-kill experiment 
task path: /utils/runtime/update_chaos_result_resource.yml:13
details => {"changed": true, "cmd": "kubectl apply -f /utils/runtime/chaos-result.yml -n shubham", "delta": "0:00:00.743588", "end": "2020-03-29 11:37:46.793624", "failed_when_result": false, "rc": 0, "start": "2020-03-29 11:37:46.050036", "stderr": "", "stderr_lines": [], "stdout": "chaosresult.litmuschaos.io/engine-container-kill configured", "stdout_lines": ["chaosresult.litmuschaos.io/engine-container-kill configured"]}
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-29 11:37:47.393703 Step: [Status]: Checking whether application pods are in running state 
task path: /utils/common/status_app_pod.yml:2
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n shubham -l run=nginx -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.356684", "end": "2020-03-29 11:37:47.374453", "rc": 0, "start": "2020-03-29 11:37:47.017769", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-03-29 11:37:47.939605 Step: [Status]: Checking whether application containers are in running state 
task path: /utils/common/status_app_pod.yml:9
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pod -n shubham -l run=nginx --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.367695", "end": "2020-03-29 11:37:47.917400", "rc": 0, "start": "2020-03-29 11:37:47.549705", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-29 11:37:48.410040 Step: [Event]: Obtain the current date in ISO8601 format 
task path: /utils/common/generate-kubernetes-chaos-events.yml:1
details => {"changed": true, "cmd": "date --iso-8601=seconds", "delta": "0:00:00.289960", "end": "2020-03-29 11:37:48.393819", "rc": 0, "start": "2020-03-29 11:37:48.103859", "stderr": "", "stderr_lines": [], "stdout": "2020-03-29T11:37:48+00:00", "stdout_lines": ["2020-03-29T11:37:48+00:00"]}

2020-03-29 11:37:48.452573 Step: set_fact 
task path: /utils/common/generate-kubernetes-chaos-events.yml:5
details => {"ansible_facts": {"ts": "2020-03-29T11:37:48Z"}, "changed": false}

2020-03-29 11:37:49.051333 Step: [Event]: Creating a PreChaosCheck event in the chaosengine 
task path: /utils/common/generate-kubernetes-chaos-events.yml:8
details => {"changed": true, "method": "patch", "result": {"apiVersion": "v1", "count": 1, "eventTime": null, "firstTimestamp": "2020-03-29T11:37:48Z", "involvedObject": {"apiVersion": "litmuschaos.io/v1alpha1", "kind": "ChaosEngine", "name": "engine", "namespace": "shubham", "uid": "ad029ce5-71b1-11ea-91b9-42010a800049"}, "kind": "Event", "lastTimestamp": "2020-03-29T11:37:48Z", "message": "AUT is Running successfully", "metadata": {"creationTimestamp": "2020-03-29T10:23:31Z", "name": "event-\"PreChaosCheck\"", "namespace": "shubham", "resourceVersion": "708", "selfLink": "/api/v1/namespaces/shubham/events/event-%22PreChaosCheck%22", "uid": "569d5193-71a7-11ea-91b9-42010a800049"}, "reason": "PreChaosCheck", "reportingComponent": "", "reportingInstance": "", "source": {"component": "container-kill-bxtsz1-wg6xg"}, "type": "Normal"}}
included: /chaoslib/pumba/pod_failure_by_sigkill.yml for 127.0.0.1

2020-03-29 11:37:49.637104 Step: Select the app pod 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:3
details => {"changed": true, "cmd": "kubectl get pod -l run=nginx -n shubham -o=custom-columns=:metadata.name --no-headers | shuf | head -1", "delta": "0:00:00.388032", "end": "2020-03-29 11:37:49.620694", "rc": 0, "start": "2020-03-29 11:37:49.232662", "stderr": "", "stderr_lines": [], "stdout": "nginx-6546cd7545-qbmsk", "stdout_lines": ["nginx-6546cd7545-qbmsk"]}

2020-03-29 11:37:49.679745 Step: Record app pod name 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:12
details => {"ansible_facts": {"app_pod": "nginx-6546cd7545-qbmsk"}, "changed": false}

2020-03-29 11:37:50.181539 Step: Identify the application node 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:17
details => {"changed": true, "cmd": "kubectl get pod nginx-6546cd7545-qbmsk -n shubham --no-headers -o custom-columns=:spec.nodeName", "delta": "0:00:00.369140", "end": "2020-03-29 11:37:50.164459", "rc": 0, "start": "2020-03-29 11:37:49.795319", "stderr": "", "stderr_lines": [], "stdout": "gke-shubham-test-default-pool-c91a8a24-762n", "stdout_lines": ["gke-shubham-test-default-pool-c91a8a24-762n"]}

2020-03-29 11:37:50.226275 Step: set_fact 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:25
details => {"ansible_facts": {"app_node": "gke-shubham-test-default-pool-c91a8a24-762n"}, "changed": false}

2020-03-29 11:37:50.651945 Step: Generate a run id if not passed from the engine/experiment 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:29
details => {"changed": true, "cmd": "echo $(mktemp) | cut -d '.' -f 2 | cut -c -6", "delta": "0:00:00.298041", "end": "2020-03-29 11:37:50.637358", "rc": 0, "start": "2020-03-29 11:37:50.339317", "stderr": "", "stderr_lines": [], "stdout": "zOK02l", "stdout_lines": ["zOK02l"]}

2020-03-29 11:37:50.694867 Step: set_fact 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:33
details => {"ansible_facts": {"run_id": "zok02l"}, "changed": false}

2020-03-29 11:37:50.796286 Step: Record the app_container 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:52
details => {"ansible_facts": {"a_container": "nginx"}, "changed": false}

2020-03-29 11:37:51.026624 Step: Patch the chaoslib image 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:62
details => {"changed": true, "checksum": "8ae1ca176955908fffcd24d94e10827530c36b27", "dest": "/chaoslib/pumba/pumba_kube.yml", "gid": 0, "group": "root", "md5sum": "230d6393ffef4686cb349ca1b3784509", "mode": "0644", "owner": "root", "size": 807, "src": "/root/.ansible/tmp/ansible-tmp-1585481870.86-214132003834771/source", "state": "file", "uid": 0}
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-29 11:37:51.502069 Step: [Event]: Obtain the current date in ISO8601 format 
task path: /utils/common/generate-kubernetes-chaos-events.yml:1
details => {"changed": true, "cmd": "date --iso-8601=seconds", "delta": "0:00:00.303953", "end": "2020-03-29 11:37:51.485601", "rc": 0, "start": "2020-03-29 11:37:51.181648", "stderr": "", "stderr_lines": [], "stdout": "2020-03-29T11:37:51+00:00", "stdout_lines": ["2020-03-29T11:37:51+00:00"]}

2020-03-29 11:37:51.545060 Step: set_fact 
task path: /utils/common/generate-kubernetes-chaos-events.yml:5
details => {"ansible_facts": {"ts": "2020-03-29T11:37:51Z"}, "changed": false}

2020-03-29 11:37:52.073008 Step: [Event]: Creating a ChaosInject event in the chaosengine 
task path: /utils/common/generate-kubernetes-chaos-events.yml:8
details => {"changed": true, "method": "patch", "result": {"apiVersion": "v1", "count": 1, "eventTime": null, "firstTimestamp": "2020-03-29T11:37:51Z", "involvedObject": {"apiVersion": "litmuschaos.io/v1alpha1", "kind": "ChaosEngine", "name": "engine", "namespace": "shubham", "uid": "ad029ce5-71b1-11ea-91b9-42010a800049"}, "kind": "Event", "lastTimestamp": "2020-03-29T11:37:51Z", "message": "Injecting container-kill chaos on nginx container of nginx-6546cd7545-qbmsk pod", "metadata": {"creationTimestamp": "2020-03-29T10:23:34Z", "name": "event-\"ChaosInject\"", "namespace": "shubham", "resourceVersion": "709", "selfLink": "/api/v1/namespaces/shubham/events/event-%22ChaosInject%22", "uid": "58655090-71a7-11ea-91b9-42010a800049"}, "reason": "ChaosInject", "reportingComponent": "", "reportingInstance": "", "source": {"component": "container-kill-bxtsz1-wg6xg"}, "type": "Normal"}}

2020-03-29 11:37:52.566973 Step: Record restartCount 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:80
details => {"changed": true, "cmd": "kubectl get pod nginx-6546cd7545-qbmsk -n shubham -o=jsonpath='{.status.containerStatuses[?(@.name==\"nginx\")].restartCount}'", "delta": "0:00:00.358758", "end": "2020-03-29 11:37:52.550528", "rc": 0, "start": "2020-03-29 11:37:52.191770", "stderr": "", "stderr_lines": [], "stdout": "4", "stdout_lines": ["4"]}
included: /utils/common/create_chaos_pod.yml for 127.0.0.1 => (item=1)
included: /utils/common/create_chaos_pod.yml for 127.0.0.1 => (item=2)
included: /utils/common/create_chaos_pod.yml for 127.0.0.1 => (item=3)

2020-03-29 11:37:53.257355 Step: Creating the chaos pod in shubham namespace 
task path: /utils/common/create_chaos_pod.yml:3
details => {"changed": true, "cmd": "kubectl apply -f /chaoslib/pumba/pumba_kube.yml -n shubham", "delta": "0:00:00.469731", "end": "2020-03-29 11:37:53.233079", "rc": 0, "start": "2020-03-29 11:37:52.763348", "stderr": "", "stderr_lines": [], "stdout": "pod/pumba-sig-kill-zok02l created", "stdout_lines": ["pod/pumba-sig-kill-zok02l created"]}
FAILED - RETRYING: Check for running status of the chaos pod (60 retries left).

2020-03-29 11:37:56.412266 Step: Check for running status of the chaos pod 
task path: /utils/common/create_chaos_pod.yml:10
details => {"attempts": 2, "changed": true, "cmd": "kubectl get pods -l name=pumba-sig-kill-zok02l -n shubham -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.415185", "end": "2020-03-29 11:37:56.392642", "rc": 0, "start": "2020-03-29 11:37:55.977457", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-03-29 11:37:56.517923 Step: Set pod_run_status to Running 
task path: /utils/common/create_chaos_pod.yml:41
details => {"ansible_facts": {"pod_run_status": "Running"}, "changed": false}

2020-03-29 11:38:17.058573 Step: Wait till the total chaos duration 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:101
details => {"changed": false, "elapsed": 20, "path": null, "port": null, "search_regex": null, "state": "started"}

2020-03-29 11:38:21.475552 Step: Tear down pumba infrastructure 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:107
details => {"changed": true, "cmd": "kubectl delete -f /chaoslib/pumba/pumba_kube.yml -n shubham", "delta": "0:00:04.271186", "end": "2020-03-29 11:38:21.455266", "rc": 0, "start": "2020-03-29 11:38:17.184080", "stderr": "", "stderr_lines": [], "stdout": "pod \"pumba-sig-kill-zok02l\" deleted", "stdout_lines": ["pod \"pumba-sig-kill-zok02l\" deleted"]}

2020-03-29 11:38:21.982549 Step: Confirm that the pumba pod is deleted successfully 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:113
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -l name=pumba-sig-kill-zok02l --no-headers -n shubham", "delta": "0:00:00.368105", "end": "2020-03-29 11:38:21.963764", "rc": 0, "start": "2020-03-29 11:38:21.595659", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}
FAILED - RETRYING: Verify restartCount (90 retries left).
FAILED - RETRYING: Verify restartCount (89 retries left).
FAILED - RETRYING: Verify restartCount (88 retries left).
FAILED - RETRYING: Verify restartCount (87 retries left).
FAILED - RETRYING: Verify restartCount (86 retries left).
FAILED - RETRYING: Verify restartCount (85 retries left).
FAILED - RETRYING: Verify restartCount (84 retries left).
FAILED - RETRYING: Verify restartCount (83 retries left).
FAILED - RETRYING: Verify restartCount (82 retries left).
FAILED - RETRYING: Verify restartCount (81 retries left).
FAILED - RETRYING: Verify restartCount (80 retries left).
FAILED - RETRYING: Verify restartCount (79 retries left).
FAILED - RETRYING: Verify restartCount (78 retries left).
FAILED - RETRYING: Verify restartCount (77 retries left).
FAILED - RETRYING: Verify restartCount (76 retries left).
FAILED - RETRYING: Verify restartCount (75 retries left).
FAILED - RETRYING: Verify restartCount (74 retries left).
FAILED - RETRYING: Verify restartCount (73 retries left).
FAILED - RETRYING: Verify restartCount (72 retries left).
FAILED - RETRYING: Verify restartCount (71 retries left).
FAILED - RETRYING: Verify restartCount (70 retries left).
FAILED - RETRYING: Verify restartCount (69 retries left).
FAILED - RETRYING: Verify restartCount (68 retries left).

2020-03-29 11:39:19.147325 Step: Verify restartCount 
task path: /chaoslib/pumba/pod_failure_by_sigkill.yml:129
details => {"attempts": 24, "changed": true, "cmd": "kubectl get pod nginx-6546cd7545-qbmsk -n shubham -o=jsonpath='{.status.containerStatuses[?(@.name==\"nginx\")].restartCount}'", "delta": "0:00:00.366604", "end": "2020-03-29 11:39:19.128726", "rc": 0, "start": "2020-03-29 11:39:18.762122", "stderr": "", "stderr_lines": [], "stdout": "5", "stdout_lines": ["5"]}
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-29 11:39:19.742643 Step: [Status]: Checking whether application pods are in running state 
task path: /utils/common/status_app_pod.yml:2
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n shubham -l run=nginx -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.392101", "end": "2020-03-29 11:39:19.712830", "rc": 0, "start": "2020-03-29 11:39:19.320729", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-03-29 11:39:20.254077 Step: [Status]: Checking whether application containers are in running state 
task path: /utils/common/status_app_pod.yml:9
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pod -n shubham -l run=nginx --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.367860", "end": "2020-03-29 11:39:20.234845", "rc": 0, "start": "2020-03-29 11:39:19.866985", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-29 11:39:20.751384 Step: [Event]: Obtain the current date in ISO8601 format 
task path: /utils/common/generate-kubernetes-chaos-events.yml:1
details => {"changed": true, "cmd": "date --iso-8601=seconds", "delta": "0:00:00.303243", "end": "2020-03-29 11:39:20.735616", "rc": 0, "start": "2020-03-29 11:39:20.432373", "stderr": "", "stderr_lines": [], "stdout": "2020-03-29T11:39:20+00:00", "stdout_lines": ["2020-03-29T11:39:20+00:00"]}

2020-03-29 11:39:20.796847 Step: set_fact 
task path: /utils/common/generate-kubernetes-chaos-events.yml:5
details => {"ansible_facts": {"ts": "2020-03-29T11:39:20Z"}, "changed": false}

2020-03-29 11:39:21.313896 Step: [Event]: Creating a PostChaosCheck event in the chaosengine 
task path: /utils/common/generate-kubernetes-chaos-events.yml:8
details => {"changed": true, "method": "patch", "result": {"apiVersion": "v1", "count": 1, "eventTime": null, "firstTimestamp": "2020-03-29T11:39:20Z", "involvedObject": {"apiVersion": "litmuschaos.io/v1alpha1", "kind": "ChaosEngine", "name": "engine", "namespace": "shubham", "uid": "ad029ce5-71b1-11ea-91b9-42010a800049"}, "kind": "Event", "lastTimestamp": "2020-03-29T11:39:20Z", "message": "AUT is Running successfully", "metadata": {"creationTimestamp": "2020-03-29T11:01:24Z", "name": "event-\"PostChaosCheck\"", "namespace": "shubham", "resourceVersion": "717", "selfLink": "/api/v1/namespaces/shubham/events/event-%22PostChaosCheck%22", "uid": "a180530f-71ac-11ea-91b9-42010a800049"}, "reason": "PostChaosCheck", "reportingComponent": "", "reportingInstance": "", "source": {"component": "container-kill-bxtsz1-wg6xg"}, "type": "Normal"}}

2020-03-29 11:39:21.359489 Step: set_fact 
task path: /experiments/generic/container_kill/container_kill_ansible_logic.yml:89
details => {"ansible_facts": {"flag": "Pass"}, "changed": false}
included: /utils/runtime/getting_failure_step.yml for 127.0.0.1

2020-03-29 11:39:21.636226 Step: [Failure-Detection]: Recording the name of failed step 
task path: /utils/runtime/getting_failure_step.yml:44
details => {"ansible_facts": {"failStep": "N/A"}, "changed": false}
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-29 11:39:22.114135 Step: [Event]: Obtain the current date in ISO8601 format 
task path: /utils/common/generate-kubernetes-chaos-events.yml:1
details => {"changed": true, "cmd": "date --iso-8601=seconds", "delta": "0:00:00.288328", "end": "2020-03-29 11:39:22.098418", "rc": 0, "start": "2020-03-29 11:39:21.810090", "stderr": "", "stderr_lines": [], "stdout": "2020-03-29T11:39:22+00:00", "stdout_lines": ["2020-03-29T11:39:22+00:00"]}

2020-03-29 11:39:22.161345 Step: set_fact 
task path: /utils/common/generate-kubernetes-chaos-events.yml:5
details => {"ansible_facts": {"ts": "2020-03-29T11:39:22Z"}, "changed": false}

2020-03-29 11:39:22.669252 Step: [Event]: Creating a Summary event in the chaosengine 
task path: /utils/common/generate-kubernetes-chaos-events.yml:8
details => {"changed": true, "method": "patch", "result": {"apiVersion": "v1", "count": 1, "eventTime": null, "firstTimestamp": "2020-03-29T11:39:22Z", "involvedObject": {"apiVersion": "litmuschaos.io/v1alpha1", "kind": "ChaosEngine", "name": "engine", "namespace": "shubham", "uid": "ad029ce5-71b1-11ea-91b9-42010a800049"}, "kind": "Event", "lastTimestamp": "2020-03-29T11:39:22Z", "message": "container-kill Experiment Passed!", "metadata": {"creationTimestamp": "2020-03-29T10:23:36Z", "name": "event-\"Summary\"", "namespace": "shubham", "resourceVersion": "718", "selfLink": "/api/v1/namespaces/shubham/events/event-%22Summary%22", "uid": "59e10420-71a7-11ea-91b9-42010a800049"}, "reason": "Summary", "reportingComponent": "", "reportingInstance": "", "source": {"component": "container-kill-bxtsz1-wg6xg"}, "type": "Normal"}}
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1

2020-03-29 11:39:23.265415 Step: [Result]: Update the chaos result CR to reflect EOT (End of Test) 
task path: /utils/runtime/update_chaos_result_resource.yml:23
details => {"changed": true, "checksum": "8e34016e18438e5b1e51291d7760232a3093758b", "dest": "/utils/runtime/chaos-result.yml", "gid": 0, "group": "root", "md5sum": "a66a4e6ecf8badb4762867705291cd4f", "mode": "0644", "owner": "root", "size": 329, "src": "/root/.ansible/tmp/ansible-tmp-1585481962.82-134540694649967/source", "state": "file", "uid": 0}

2020-03-29 11:39:23.897192 Step: [The END]: Apply the chaos result CR for container-kill experiment 
task path: /utils/runtime/update_chaos_result_resource.yml:33
details => {"changed": true, "cmd": "kubectl apply -f /utils/runtime/chaos-result.yml -n shubham", "delta": "0:00:00.487166", "end": "2020-03-29 11:39:23.877429", "failed_when_result": false, "rc": 0, "start": "2020-03-29 11:39:23.390263", "stderr": "", "stderr_lines": [], "stdout": "chaosresult.litmuschaos.io/engine-container-kill configured", "stdout_lines": ["chaosresult.litmuschaos.io/engine-container-kill configured"]}
META: ran handlers
META: ran handlers

2020-03-29 11:39:23.902305 ************ RELAX, EXPERIMENT ENDS! ************ 
127.0.0.1                  : ok=56   changed=27   unreachable=0    failed=0   


```

**Comment by ispeakc0de** [comment]
**Job logs if unable to come to running state**
[job-logs.txt](https://github.com/litmuschaos/litmus/files/4398865/job-logs.txt)

File: chaoslib/pumba/pod_failure_by_sigkill.yml [hunk 1]
```diff
@@ -93,34 +85,43 @@
...
+    # If pumba pod won't come to running state after three reties, It will fail the execution.
+    - include_tasks: /utils/common/create_chaos_pod.yml
+      vars:
+        pod_ns: "{{ namespace }}"
+        c_path: "/chaoslib/pumba/pumba_kube.yml"
+      with_sequence: start=1 end=3
+    
+    # Failing the execution, If pumba pod won't come to running state after three retries.
+    - fail:
+        msg: "pod_failure_by_sigkill lib failed, Unable to create pumba pod"
```
Comment (ksatchit) [inline chaoslib/pumba/pod_failure_by_sigkill.yml]: Unable to create as pumba couldn't be scheduled on the app node.

File: utils/common/create_chaos_pod.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        kubectl get pods -l name=pumba-sig-kill-{{ run_id }} -n {{ pod_ns }} -o custom-columns=:.status.phase --no-headers
+      register: pod_result
+      until: "((pod_result.stdout.split()|unique)|length) == 1 and 'Running' in pod_result.stdout"
+      delay: 2
+      retries: 90
+      ignore_errors: true
+
+    - block:
+
+        - name: Delete the chaos pod, If not comming to running state
```
Comment (ksatchit) [inline utils/common/create_chaos_pod.yml]: Delete the pumba pod, if it isn't scheduled successfully

File: chaoslib/pumba/pod_failure_by_sigkill.yml [hunk 3]
```diff
@@ -93,34 +85,43 @@
...
 
-    - name: Setup pumba chaos infrastructure
-      shell: >
-        kubectl apply -f /chaoslib/pumba/pumba_kube.yml
-        -n {{ namespace }}
-      args:
-        executable: /bin/bash
-      register: pumba_job_result
+    # It creates the pumba pod and retries three times for the creation of pumba pod
+    # If pumba pod won't come to running state after three reties, It will fail the execution.
```
Comment (ksatchit) [inline chaoslib/pumba/pod_failure_by_sigkill.yml]:
```suggestion
    # Pumba pods creation is attempted for a total of 3 times, if it is not immediately schedulable due to transient node conditions
    # If the pumba pod is not schedulable across these 3 tries, the experiment is failed with message indicating improper cluster state.
```

File: chaoslib/pumba/pod_failure_by_sigkill.yml [hunk 4]
```diff
@@ -145,18 +146,18 @@
...
+            kubectl get pods -l name=pumba-sig-kill-{{ run_id }} --no-headers -n {{ namespace }}
           args:
             executable: /bin/bash
           register: result
           until: "'No resources found' in result.stderr"
           delay: 2
-          retries: 60
-      when: "pumba_job_result is defined"
+          retries: 90
+      when: "(pod_run_status is defined and pod_run_status == 'Running') and pumba_pod_result is defined"
```
Comment (ksatchit) [inline chaoslib/pumba/pod_failure_by_sigkill.yml]: Is this mandatory?

Comment (ksatchit) [reply chaoslib/pumba/pod_failure_by_sigkill.yml]: Ok it is because of invalid state of pod?

Comment (ispeakc0de) [reply chaoslib/pumba/pod_failure_by_sigkill.yml]: Yes, If the pod_run_status is not `Running` then the util itself deleted the pumba pod.

File: utils/common/create_chaos_pod.yml (hunk 2 above)
Comment (ispeakc0de) [reply utils/common/create_chaos_pod.yml]: I am using chaos pod to make it generic, as this util will be used by other experiments too.

File: chaoslib/pumba/pod_failure_by_sigkill.yml (hunk 4 above)
Comment (ksatchit) [reply chaoslib/pumba/pod_failure_by_sigkill.yml]: cool

File: chaoslib/pumba/pod_failure_by_sigkill.yml [hunk 5]
```diff
@@ -92,35 +84,44 @@
...
+          shell: >
+            kubectl get pods -l name=pumba-sig-kill-{{ run_id }} --no-headers -n {{ namespace }}
+          args:
+            executable: /bin/bash
+          register: result
+          until: "'No resources found' in result.stderr"
+          delay: 2
+          retries: 90
+
+      when: "pod_run_status is defined and pod_run_status == 'Running'"
```
Comment (uditgaurav) [inline chaoslib/pumba/pod_failure_by_sigkill.yml]: Can we use `pod_status` in place of `pod_run_status` everywhere?

**Pull request merged**